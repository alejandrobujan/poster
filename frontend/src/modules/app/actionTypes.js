export const ERROR = 'poster/common/error';