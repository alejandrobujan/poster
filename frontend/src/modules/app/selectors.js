const getModuleState = state => state.app;

export const getError = state => getModuleState(state).error;

export const getFirstSearch = state => getModuleState(state).firstSearch;