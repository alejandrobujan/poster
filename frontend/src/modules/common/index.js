export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";