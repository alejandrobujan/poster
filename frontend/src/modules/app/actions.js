import * as actionTypes from './actionTypes';

export const error = error => ({
    type: actionTypes.ERROR,
    error
});

export const setFirstSearch = firstSearch => ({
    type: actionTypes.SET_FIRST_SEARCH,
    firstSearch
});