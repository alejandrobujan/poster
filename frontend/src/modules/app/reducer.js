import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    error: null,
    firstSearch: true
};

const error = (state = initialState.error, action) => {

    switch (action.type) {

        case actionTypes.ERROR:
            return action.error;

        default:
            return state;

    }

};    

const firstSearch = (state = initialState.firstSearch, action) => {

    switch (action.type) {

        case actionTypes.SET_FIRST_SEARCH:
            return action.firstSearch;

        default:
            return state;

    }

};    

const reducer = combineReducers({
    error,
    firstSearch
});

export default reducer;