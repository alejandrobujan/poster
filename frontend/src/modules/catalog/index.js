import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


