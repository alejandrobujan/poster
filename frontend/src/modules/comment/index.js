import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};