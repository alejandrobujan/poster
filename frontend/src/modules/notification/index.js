import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


