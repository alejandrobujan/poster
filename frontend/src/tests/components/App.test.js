import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import { App } from "../../modules/app";
import store from "../../store";

describe("App", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <MemoryRouter>
                        <App/>
                    </MemoryRouter>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


    it("no posts", async () => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );


        /*const backButton = screen.getByRole('alert', {name: 'No posts found'});
        await waitFor(() => {
            screen.getByText('No posts found'); //NON FURRULA
        })*/
        

    });

    it("appears buttons back and next", async () => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );

        /*const backButton = screen.getByRole('button', {name: 'Back'});
        const nextButton = screen.getByRole('button', {name: 'Next'});*/
    });

    /*
    const inputText = screen.getByLabelText(/input some text/i);
    fireEvent.change(inputText, {
      target: { value: "Sreenzragnf qr qrfraibyirzragb" },
    });
    const calculateButton = screen.getByText(/calculate/i);
    fireEvent.click(calculateButton);
    await waitFor(() => {
      screen.getByText(/Ferramentas de desenvolvemento/);
    });
    */
});

/*
fireEvent.change(inputText, {
            target: { value: "Sreenzragnf qr qrfraibyirzragb" },
        });
        const calculateButton = screen.getByText(/calculate/i);
        fireEvent.click(calculateButton);
        await waitFor(() => {
            screen.getByText(/Ferramentas de desenvolvemento/);
        });
*/
