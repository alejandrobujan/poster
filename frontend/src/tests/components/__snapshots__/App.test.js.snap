// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light border"
  >
    <a
      className="navbar-brand"
      href="/poster"
      onClick={[Function]}
    >
      <img
        alt="Poster"
        height="50px"
        src="/poster/assets/logo.png"
      />
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      />
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/poster/users/login"
            id="loginLink"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    style={
      Object {
        "marginBottom": "100px",
      }
    }
  />
  <div
    className="fixed-bottom bg-light border-top pt-3"
  >
    <footer>
      <p
        className="text-center"
      >
        © 2023 Poster Group - Development Tools - University of A Coruña
      </p>
    </footer>
  </div>
</div>
`;
